src/lib/utils/DY.Lib.SplitFunction.fst

```diff
@@ -398,3 +398,28 @@ let rec memP_mk_dependent_tagged_local_funs #tag_set_t #local_fun_t l tag_set lo
   | [] -> ()
   | (x,y)::t ->
     memP_mk_dependent_tagged_local_funs t tag_set local_fun
+
+/// Normalize pre-condition and post-conditions from correctness lemmas derived from `mk_global_fun_correct`
+/// to save some boilerplate when using this framework
+
+val get_fv_name: FStar.Tactics.term -> FStar.Tactics.Tac unit
+let get_fv_name t =
+  let open FStar.Tactics in
+  match t with
+  | Tv_FVar fv -> (
+    exact (Tv_Const (C_String (implode_qn (inspect_fv fv))))
+  )
+  | _ -> fail ("get_fv_name: not a free variable: " ^ (term_to_string t))
+
+val do_boilerplate:
```

Comment:
I would suggest renaming to `do_split_function_boilerplate` or `do_split_boilerplate` or something similar --- it looks like, while nothing in this explicitly relates to the split functions, it is purpose-built for it. Maybe someone else has a better idea for the name?

It appears that the general purpose of this is to take in an association list (without repeats) and show generically that if every entry in the association list satisfies `p`, then the overall list satisfies `for_allP q` (and expand that `for_allP` to a chain of `/\`). A good name for this doesn't immediately come to mind, but unless we expect to use it outside of the scope of split functions, I think it's fine to have a name reflecting that intended use for now.